pkg/resources/management.cattle.io/v3/project/validator_test.go

```diff
@@ -929,6 +930,417 @@ func TestProjectValidation(t *testing.T) {
 	}
 }
 
+func TestProjectContainerDefaultLimitsValidation(t *testing.T) {
+	t.Parallel()
+	type testState struct {
+		clusterCache *fake.MockNonNamespacedCacheInterface[*v3.Cluster]
+	}
+	tests := []struct {
+		name        string
+		operation   v1.Operation
+		newProject  *v3.Project
+		oldProject  *v3.Project
+		wantAllowed bool
+		wantErr     bool
+	}{
+		{
+			name: "container default resource limit specified cpu and memory request and limits",
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "1m",
+						LimitsCPU:      "20m",
+						RequestsMemory: "1Mi",
+						LimitsMemory:   "20Mi",
+					},
+				},
+			},
```

Comment:
Nit:

One benefit of splitting out the tests was that you didn't need to define the whole project each time, you could instead just define the ContainerDefault resource limit, like the below:

```suggestion
			newLimit: &v3.ContainerResourceLimit{
						RequestsCPU:    "1m",
						LimitsCPU:      "20m",
						RequestsMemory: "1Mi",
						LimitsMemory:   "20Mi",
					},
```

Reply:
Yup, simplified.